File: leetcode/0231_power-of-two/main.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPowerOfTwo = function(n) {
  return n > 0 && (n & (n - 1)) === 0;
};

// 测试
var data = [
  { n: -1, result: false },
  { n: 0, result: false },
  { n: 1, result: true },
  { n: 16, result: true },
  { n: 218, result: false }
];
for (var i of data) {
  var result = isPowerOfTwo(i.n);
  console.log(result == i.result ? 'ok' : 'failed');
}

File: leetcode/0231_power-of-two/main.c
#include <stdio.h>

int isPowerOfTwo(int n)
{
    return n > 0 && (n & (n - 1)) == 0;
}

/* 测试 */
int main(int argc, char const *argv[])
{
    int nums[] = {-1, 0, 1, 16, 218};
    int result;
    for (int i = 0; i < 5; ++i) {
        result = isPowerOfTwo(nums[i]);
        printf("%s\n", result ? "true" : "false");
    }
    return 0;
}
/*
false
false
true
true
false
*/
